rust/otap-dataflow/crates/otap/src/experimental/azure_monitor_exporter/auth.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use azure_core::credentials::{AccessToken, TokenCredential};
+use azure_core::time::OffsetDateTime;
+use azure_identity::{
+    DeveloperToolsCredential, DeveloperToolsCredentialOptions, ManagedIdentityCredential,
+    ManagedIdentityCredentialOptions, UserAssignedId,
+};
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use crate::experimental::azure_monitor_exporter::config::{AuthConfig, AuthMethod};
+
+#[derive(Clone)]
+// TODO - Remove print statements
+#[allow(clippy::print_stdout)]
+pub struct Auth {
+    credential: Arc<dyn TokenCredential>,
+    scope: String,
+    // Thread-safe shared token cache
+    cached_token: Arc<RwLock<Option<AccessToken>>>,
+}
+
+// TODO: Remove print_stdout after logging is set up
+#[allow(clippy::print_stdout)]
```

Comment:
I will stake out a position and ask @cijothomas for an opinion too about plain-old-logging APIs with message formatting like `println!("string {variable}")` where `variable` is borrowed if the statement executes.

As I've worked now with Arrow and Datafusion, here's what I find:

1. Not much logging in Arrow itself (Arrow Flight the RPC framework uses `tracing_log`)
2. DataFusion is heavily instrumented with the `log` package with `env_logger` (I know I can set RUST_LOG=debug and learn about its internals)

I like the `log` API surface. 

Now, I'm asking @lquerel for an opinion too. How would you like to see a logging SDK adopted? I think `env_logger` is great in development settings.

Reply:
`tracing` macros are what we use in OTel-Rust. It's zero cost when not-enabled. And we can setup subscribers (listeners that routes logs to destination like otel or stdout or file etc..) per thread.

(`tracing` has a `log` bridge too, so the subscriber we use for `tracing` will be able to get `log` events as well)

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/global/internal_logging.rs
OTel Rust wrote wrapper macros, so we can easily switch from `tracing` to something else, if needed.